src/bw_admin.rs

```diff
@@ -31,29 +33,50 @@ impl User {
 pub struct Client {
     url: String,
     admin_token: String,
-    cookie: Option<String>,
-    cookie_created: Option<Instant>,
+    root_cert: String,
+	cookie: Option<String>,
+    cookie_created: Option<Instant>
 }
 
 impl Client {
     /// Create new instance of client
-    pub fn new(url: String, admin_token: String) -> Client {
+    pub fn new(url: String, admin_token: String, root_cert: String) -> Client {
         Client {
             url,
             admin_token,
-            cookie: None,
-            cookie_created: None,
+            root_cert,
+			cookie: None,
+            cookie_created: None
         }
     }
 
+    fn get_http_client(&self) -> reqwest::Client {
+
+        let mut client = reqwest::Client::builder()
+            .redirect(reqwest::RedirectPolicy::none());
+
+        if !&self.root_cert.is_empty() {
+            
+            let mut buf = Vec::new();
+
+            // read a local binary DER encoded certificate
+            File::open(&self.root_cert).expect("cannot open root cert").read_to_end(&mut buf).expect("cannot read root cert");
+
+            // create a certificate
+            let cert = reqwest::Certificate::from_der(&buf).expect("could not load der certificate");
+
+			// add the root cert
+            client = client.add_root_certificate(cert);
+        }
+
+        return client.build().unwrap();
+    }
+
     /// Authenticate client
     fn auth(&mut self) -> Response {
+
         let cookie_created = Instant::now();
-        let client = reqwest::Client::builder()
-            // Avoid redirects because server will redirect to admin page after auth
-            .redirect(reqwest::RedirectPolicy::none())
-            .build()
-            .unwrap();
+        let client = &self.get_http_client();
```

Comment:
Here you use `let client = &self.get_http_client();` while below you use `let client = self.get_http_client();`. Is there a reason for this difference?

Reply:
No reason, thanks for spotting it. Fixed.